src/consensus/consensus.ml

```diff
@@ -0,0 +1,16 @@
+include State
+include Effect
+
+type t = state
+
+let with_block effect block state =
```

Comment:
I was thrown off by this variable name. I suggest `effect_handler` instead for this one, below, and `dispatch.ml`